make_prg/utils/seq_utils.py

```diff
@@ -0,0 +1,220 @@
+        for sequence in sequences:
+            if not set(sequence).issubset(cls.allowed_bases):
+                raise SequenceCurationError(
+                    "A slice of a sequence has a disallowed base.\n"
+                    f"Allowed bases: {cls.allowed_bases}.\n"
+                    f"Sequence: {sequence}\n"
+                    f"Redo sequence curation.\n"
+                )
+
+    @classmethod
+    def check_all_sequences_are_composed_of_ACGT(cls, sequences: List[str]):
+        for sequence in sequences:
+            sequence_is_composed_of_ACGT_only = set(sequence).issubset(cls.standard_bases)
+            assert sequence_is_composed_of_ACGT_only, f"Sequence ({sequence}) should be composed of ACTG only."
+
+    @classmethod
+    def get_expanded_sequences(cls, sequences: List[str]) -> Sequences:
+        """
+        Expand sequences in the given list of sequences, following the translation table in SequenceExpander.iupac.
+        It does the following steps:
+            1. Check that we don't have disallowed bases;
+            2. Remove sequences with N;
+            3. Duplicate sequences containing RYKMSW, replacing with AGCT alternatives;
+        Note 1: The sequences are deliberately returned in the order they are received.
+        Note 2: Returned sequences are composed of ACGT only
+        """
+
+        cls.check_if_there_is_sequence_with_disallowed_bases(sequences)
+
+        expanded_seqs = []
+        expanded_set = set()
+        deduplicated_gapless_seqs = remove_duplicates(sequences)
+        for seq in deduplicated_gapless_seqs:
+            if "N" in seq:
+                continue
+
+            alternatives = [cls.iupac[base] for base in seq]
+            for tuple_product in itertools.product(*alternatives):
+                expanded_str = "".join(tuple_product)
+                if expanded_str not in expanded_set:
+                    expanded_set.add(expanded_str)
+                    expanded_seqs.append(expanded_str)
+
+        all_sequences_contained_N = len(expanded_seqs) == 0
+        if all_sequences_contained_N:
+            raise SequenceCurationError(
+                "All sequences in this slice contained N. Redo sequence curation.\n"
+                f"Sequences: {sequences}"
+            )
+
+        cls.check_all_sequences_are_composed_of_ACGT(expanded_seqs)
+        return expanded_seqs
+
+    @classmethod
+    def get_expanded_sequences_from_MSA(cls, alignment: MSA) -> Sequences:
+        gapless_seqs = list(map(ungap, get_alignment_seqs(alignment)))
+        return cls.get_expanded_sequences(gapless_seqs)
+
+
+def align(seq1: str, seq2: str,
+          match_score: float = 2,
+          mismatch_score: float = -0.9,
+          gap_open_score: float = -1.1,
+          gap_extend_score: float = -1) -> Tuple[str, str]:
+    alignment = pairwise2.align.globalms(
+        seq1, seq2, match_score, mismatch_score, gap_open_score, gap_extend_score, one_alignment_only=True
+    )
```

Comment:
I remember @mbhall88  or martin saying BioPython's pairwise2 is v. slow- should we replace it with `edlib`?